app/src/main/java/org/apache/taverna/mobile/ui/tutorial/TutorialActivity.java

```diff
@@ -0,0 +1,203 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.text.Html;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import org.apache.taverna.mobile.R;
+import org.apache.taverna.mobile.data.local.PreferencesHelper;
+import org.apache.taverna.mobile.ui.FlashScreenActivity;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+
+public class TutorialActivity extends AppCompatActivity {
+
+    @BindView(R.id.slide_pager)
+    ViewPager slidePager;
+
+    @BindView(R.id.layoutDots)
+    LinearLayout dotsLayout;
+
+    @BindView(R.id.btn_skip)
+    Button bSkip;
+
+    @BindView(R.id.btn_next)
+    Button bNext;
+
+    private TutorialSliderAdapter tutorialSliderAdapter;
+    int[] layouts;
+    private PreferencesHelper preferencesHelper;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        preferencesHelper = new PreferencesHelper(this);
+        if (!preferencesHelper.isFirstTimeLaunch()) {
+            launchFlashScreen();
+            finish();
+        }
+
+        setContentView(R.layout.activity_tutorial);
+
+        ButterKnife.bind(this);
+
+
+        if (Build.VERSION.SDK_INT >= 21) {
+            getWindow().getDecorView()
+                    .setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                            | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+        }
+
+        layouts = new int[]{
+                R.layout.tutorial_slide1,
+                R.layout.tutorial_slide2,
+                R.layout.tutorial_slide3,
+                R.layout.tutorial_slide4};
+
+        addBottomDots(0);
+
+        tutorialSliderAdapter = new TutorialSliderAdapter();
+        slidePager.setAdapter(tutorialSliderAdapter);
+        slidePager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
```

Comment:
change this line to `slidePager.addOnPageChangeListener(this);` and impelement its interface to the class.